helm/freenas-provisioner/templates/deployment.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "freenas-provisioner.fullname" . }}
+  labels:
+    {{- include "freenas-provisioner.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "freenas-provisioner.selectorLabels" . | nindent 6 }}
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        {{- include "freenas-provisioner.selectorLabels" . | nindent 8 }}
+    spec:
+    {{- with .Values.image.pullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ include "freenas-provisioner.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Chart.AppVersion }}"
```

Comment:
With this, it would be impossible to fully override the image if needed...
I would suggest letting the chart user decide whether he/she wants to use another image version (or perhaps a local snapshot version for testing).

Reply:
This seems to be the new pattern and is how `helm create` builds it. I tend to agree that letting the user override it is good. Added.